### ISSUE TITLE
APK files are not supported

### ISSUE THREAD

**Issue by TheSETJ** [thephpleague/flysystem#1092]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 1.0.8

#### Summary

I have a Laravel API for uploading file and I use S3 compatible storage.

The issue is that every `.apk` file being uploaded seems to be corrupted after saving to the storage. I dig into Laravel source and found out that `AwsS3Adapter` class handles storing files. In here `.apk` files are being guessed to be of `application/java-archive` or sometimes `application/zip` content type by `guessMimeType()` method of `Util` class. I also tried passing the content type manually giving `[ 'ContentType' => 'application/vnd.android.package-archive' ]` configuration through your package which at last will be passed as `$config` parameter to `upload()` method in `AwsS3Adapter` class, but that didn't resolve the issue.

I also tried storing the file using the `local` driver, but the issue persists for the locally stored files too. 

#### How to reproduce

- Create a project using Laravel framework and write an API route to upload file
- Use the following code to store your `.apk` file:
```
// For .apk file $options should be [ 'ContentType' => 'application/vnd.android.package-archive' ];
$file->storeAs($directory, $filename, $options);
```
- Download the file from whatever storage your using and try to open it on an android device.

**Comment by frankdejonge** [comment]
@TheSETJ can you reproduce this with Flysystem alone?

**Comment by TheSETJ** [comment]
> @TheSETJ can you reproduce this with Flysystem alone?

I didn't try it standalone though as I took a look into your files, I know you even don't have an `apk` entry in [your mime type list](https://github.com/thephpleague/flysystem/blob/f1f459132312434cd58b84c1a146eef37fc8933f/src/Util/MimeType.php#L13-L188).

**Comment by frankdejonge** [comment]
You're not providing enough information to investigate this issue. Unless you can narrow the scope or give more information, there's not much I can do.

**Comment by frankdejonge** [comment]
BTW, I'm not going on a full exploration because you spotted a missing mimet-type, that's not how this works.

**Comment by TheSETJ** [comment]
> BTW, I'm not going on a full exploration because you spotted a missing mimet-type, that's not how this works.

I really don't know how to use a package, standalone, outside any framework (because I don't have enough experience).
By the way, about the mime-type in this file, you need to add the following line to support `apk` extension:

```php
$extensionToMimeTypeMap = [
    'apk' => 'application/vnd.android.package-archive',
    // other entries
]
```

It does not need a full exploration.

**Comment by frankdejonge** [comment]
PR welcome, I have no interest in fixing this myself. I do not appreciate your tone of voice.

**Comment by TheSETJ** [comment]
> PR welcome, I have no interest in fixing this myself. I do not appreciate your tone of voice.

I didn't mean any insult. I told the truth. I'm really not very experienced with PHP and Laravel.

**Comment by TheSETJ** [comment]
After many searches, it seems it's a PHP bug which if I got it right, it's fixed in PHP 7.2:

> This is fixed in 7.2 with the libmagic upgrade

But I don't know anything about that `libmagic`. Here is the [main discussion](https://bugs.php.net/bug.php?id=74573) and how can it be used/updated in your package.
